### ISSUE TITLE
Unable to compile EnderIO

### ISSUE THREAD

**Issue by embeddedt** [SleepyTrousers/EnderIO#5533]
#### Issue Description:

I am unable to compile EnderIO from source.

#### What happens:

I receive an error about failure to locate the deobfuscated version of EnderCore.

#### What you expected to happen:

Compilation should succeed.

#### Steps to reproduce:

1. `git clone https://github.com/SleepyTrousers/EnderIO`
2. `cd EnderIO`
3. `./gradlew setupDecompWorkspace`
4. `./gradlew build`

____
#### Affected Versions (Do *not* use "latest"):

- EnderIO: 9249b7e01fd73c7bc92a2fd60e768c1ef5b39fd3
- EnderCore: 0.5.76
- Minecraft: 1.12.2
- Forge: 14.23.5.2847
- SpongeForge? N/A
- Optifine? N/A
- Single Player and/or Server? N/A

#### Your most recent log file where the issue was present: 

I removed several of the repeated warnings about FG 5 and the mappings.

```
> Configure project : 
This mapping 'snapshot_20171212' was designed for MC 1.12! Use at your own peril.
This mapping 'snapshot_20171212' was designed for MC 1.12! Use at your own peril.
WARNING: You are using an unsupported version of ForgeGradle.
Please consider upgrading to ForgeGradle 5 and helping in the efforts to get old versions working on the modern toolchain.
See https://gist.github.com/TheCurle/fe7ad3ede188cbdd15c235cc75d52d4a for more info on contributing.

> Configure project :enderio-integration-thaumcraft 
WARNING: You are using an unsupported version of ForgeGradle.
Please consider upgrading to ForgeGradle 5 and helping in the efforts to get old versions working on the modern toolchain.
See https://gist.github.com/TheCurle/fe7ad3ede188cbdd15c235cc75d52d4a for more info on contributing.

FAILURE: Build failed with an exception.

* What went wrong:
Could not resolve all files for configuration ':ap:compileClasspath'.
> Could not find EnderCore.jar (deobf.com.enderio.core:EnderCore:1.12.2-0.5.76).
  Searched in the following locations:
      http://files.minecraftforge.net/maven/deobf/com/enderio/core/EnderCore/1.12.2-0.5.76/EnderCore-1.12.2-0.5.76.jar

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.

* Get more help at https://help.gradle.org

BUILD FAILED in 3s
5 actionable tasks: 3 executed, 2 up-to-date
```

**Comment by CitiesXL2815** [comment]
oh your using outdated forge version use 2854 see if this still exist

**Comment by embeddedt** [comment]
According to your build.gradle you are still using ForgeGradle 2.3:

https://github.com/SleepyTrousers/EnderIO/blob/9249b7e01fd73c7bc92a2fd60e768c1ef5b39fd3/build.gradle#L14

As far as I know, the newer versions of Forge require ForgeGradle 3.

I should note that I am also able to compile other Forge 2847 mods like [More Refined Storage](https://github.com/tth05/morerefinedstorage) without issue.

**Comment by hypherionmc** [comment]
EnderCore is not included by default. You need to place the EnderCore file from our maven into the "libs" folder, or include it as a dependency in build.gradle. Somewhere is a guide on how to properly setup the workspace, but I can't find it now. That's the way I set it up for myself

Maven is https://maven.tterrag.com

**Comment by hypherionmc** [comment]
Also just curious, why are you wanting to compile it yourself? We have [dev builds](https://ci.tterrag.com/job/EnderIO-Modules/job/EnderIO-Hourly/) for everything that gets comitted here. If you are wanting to help port it, then you should really join our discord so you can co-ordinate with everyone else already working on porting

**Comment by embeddedt** [comment]
> EnderCore is not included by default. You need to place the EnderCore file from our maven into the "libs" folder, or include it as a dependency in build.gradle.

Thanks. That was the step I had missed.

(The Forge Maven URL also needs to be adjusted; I'll send a PR for that.)

> Also just curious, why are you wanting to compile it yourself?

I still use 1.12, mainly for performance reasons; the newer renderer is a lot slower even with Sodium/other performance mods. As such, I'm often needing to patch various issues in mods. Most of the time I just use mixins, as only a few developers are willing to accept fixes for 1.12, even if community-contributed. EnderIO appears to still be somewhat maintained for 1.12, so I'd rather provide a direct fix upstream than use mixins.
